server/services/auditService.ts

```diff
@@ -347,7 +347,7 @@ export default class AuditService {
       operationId,
       who,
       service: config.apis.audit.serviceName,
-      details: JSON.stringify(details),
+      details: JSON.stringify(details, this.replaceUndefinedWithNull),
```

Comment:
`undefined` is not valid in JSON so such values were 'disappearing' from user audit messages. E.g. change establishment log when first logging in was losing `previousEstablishment: undefined`.

This fix replaces any `undefined` values that might crop up in audit messages with `null` so they do get logged.

Reply:
Nice 👍